constants/src/main/java/org/jboss/pnc/enums/BuildStatus.java

```diff
@@ -24,45 +24,58 @@
 import static org.jboss.pnc.enums.BuildProgress.PENDING;
 
 /**
- * Status of a running or finished build.
+ * Status of a running/completed Build/GroupBuild.
+ * 
+ * The status class is shared by both Build and GroupBuild dtos, which are published through UI/WebSocket.
  *
  * @author Jakub Bartecek &lt;jbartece@redhat.com&gt;
  *
  */
 public enum BuildStatus {
     /**
-     * Build completed successfully.
+     * Build/GroupBuild has completed successfully. The dependant Builds can be scheduled.
      */
     SUCCESS(FINISHED, true),
 
     /**
-     * Build failed.
+     * Build/GroupBuild has failed. The status is propagated through dependants as REJECTED_FAILED_DEPENDENCIES.
+     *
+     * A GroupBuild is reported FAILED with is at least one underlying Build with the final status of (FAILED, REJECTED,
```

Comment:
`s/with is/when/` `s/with the/has the/`? Or smething like that, it doesn't read well now.

Reply:
+:100: